cmake/extensions.cmake

```diff
@@ -355,7 +368,7 @@ function(generate_inc_file_for_target
   set(generated_target_name "gen_${basename}_${random_chars}")
 
   add_custom_target(${generated_target_name} DEPENDS ${generated_file})
-  add_dependencies(${target} ${generated_target_name})
+  generate_inc_file_for_target_with_target_name(${target} ${generated_target_name} ${source_file} ${generated_file} ${ARGN})
```

Comment:
The second argument in the invocation is

generated_target_name

but the second argument in the declaration is source_file

Reply:
There is something else fishy here, but I'm not able to tell what exactly it is.